r2-core/src/main/java/com/linkedin/r2/filter/transport/FilterChainClient.java

```diff
@@ -46,15 +52,17 @@
 {
   private final TransportClient _client;
   private final FilterChain _filters;
+  private final FilterChain _sharedFilters;
 
   /**
    * Construct a new instance by composing the specified {@link TransportClient}
    * and {@link FilterChain}.
    *
    * @param client the {@link TransportClient} to be composed.
    * @param filters the {@link FilterChain} to be composed.
+   * @param sharedFilters the {@link FilterChain} can be used by other clients.
    */
-  public FilterChainClient(TransportClient client, FilterChain filters)
+  public FilterChainClient(TransportClient client, FilterChain filters, FilterChain sharedFilters)
```

Comment:
This is a backward-incompatible change, can we introduce a new overloaded constructor and deprecate the usage of the existing constructor?

Reply:
Fixed with a flag in the filter instead.